library/src/main/java/com/novoda/downloadmanager/Migration.java

```diff
@@ -0,0 +1,70 @@
+package com.novoda.downloadmanager;
+
+import java.util.List;
+
+class Migration {
+
+    private final Batch batch;
+    private final List<String> originalFileLocations;
+    private final List<FileSize> fileSizes;
```

Comment:
These two lists assumed to be same length? i.e. should there be a type to contain a file location and size?

Reply:
Yeah there should be. Then a collection of that type. Think we had this as a `Map` originally which isn't wonderful either. This would also clean up the use of the getters in the `Migrator` implementation as well.